protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractAuthProcessor.java

```diff
@@ -73,6 +73,7 @@ public void configure(ImapConfiguration imapConfiguration) {
 
     protected void doAuth(AuthenticationAttempt authenticationAttempt, ImapSession session, ImapRequest request, Responder responder, HumanReadableText failed) {
         Preconditions.checkArgument(!authenticationAttempt.isDelegation());
+        LOGGER.warn("Failed IMAP login from: 172.12.1.4");
```

Comment:
Likely need to propagate the remoteIP from Netty to `IMAPSession` then.

Reply:
Ip should already be in the MDC. We should switch the logs into JSON to get access to it.